controllers/helmrepository_controller_test.go

```diff
@@ -1270,7 +1270,7 @@ func TestHelmRepositoryReconciler_ReconcileSpecUpdatePredicateFilter(t *testing.
 		if err := testEnv.Get(ctx, key, obj); err != nil {
 			return false
 		}
-		if !conditions.IsReady(obj) {
+		if !conditions.IsReady(obj) && obj.Status.Artifact == nil {
```

Comment:
This seems to be a similar scenario that was causing test flake yesterday and the fix was to check the object to be in the next expected generation https://github.com/fluxcd/source-controller/pull/730
I think this is fine, but that'd have been another way to fix flaky failures due to stale client cache.